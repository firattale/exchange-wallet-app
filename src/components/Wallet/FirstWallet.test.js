import React from 'react';
import { mount } from 'enzyme';
import { FirstWallet } from './FirstWallet.react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from "../../app/walletSlice"

describe('FirstWallet', () => {
    const mockAppState = {
        exchange: {
            firstCurrency: "GBP",
            firstSign: "£",
            secondCurrency: "USD",
            currencyRate: "",
            secondSign: "$"
        },
        wallet: {
            USD: 40.00,
            GBP: 50.00,
            EUR: 60.00,
            firstAmount: "",
            secondAmount: "",
            firstWalletError: null
        }
    }
    const mockProps = {
        sign: "£",
        firstAmount: 50.00,
        currency: "GBP",
        secondCurrency: "USD",
        currencyRate: ""
    }
    const mockStore = createStore(reducer, mockAppState);
    mockStore.dispatch = jest.fn();
    const getWrapper = () => mount(
        <Provider store={mockStore}>
            <FirstWallet {...mockProps} />
        </Provider>
    );

    afterEach(() => {
        jest.clearAllMocks()
    });


    it('should render correctly in "debug" mode', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    });

    it('should call dispatch func when  number inputs change', () => {
        const event = {
            target: {
                value: 1.00
            }
        }
        const wrapper = getWrapper();
        const input = wrapper.find('[data-testid="wallet-input-1"]');
        input.simulate("change", event);
        expect(mockStore.dispatch).toHaveBeenCalledTimes(3);
        expect(mockStore.dispatch).toHaveBeenCalledWith({ "payload": { "error": null }, "type": "wallet/checkFirstWalletError" });
        expect(mockStore.dispatch).toHaveBeenCalledWith({ "payload": { "firstAmount": 1 }, "type": "wallet/changeFirstAmount" });
        expect(mockStore.dispatch).toHaveBeenCalledWith({ "payload": { "secondAmount": "0.00" }, "type": "wallet/changeSecondAmount" });
    });
    it('should call dispatch func when select inputs change', () => {
        const event = {
            target: {
                value: "USD"
            }
        }
        const wrapper = getWrapper();
        const input = wrapper.find('[data-testid="wallet-select"]').at(0);
        input.simulate("change", event);
        expect(mockStore.dispatch).toHaveBeenCalledTimes(3);
        expect(mockStore.dispatch).toHaveBeenCalledWith({ "payload": { "first": "USD", "firstSign": "$" }, "type": "exchange/changeFirstCurrency" });
        expect(mockStore.dispatch).toHaveBeenCalledWith({ "payload": { "firstAmount": "" }, "type": "wallet/changeFirstAmount" });
        expect(mockStore.dispatch).toHaveBeenCalledWith({ "payload": { "secondAmount": "" }, "type": "wallet/changeSecondAmount" });

    });

});