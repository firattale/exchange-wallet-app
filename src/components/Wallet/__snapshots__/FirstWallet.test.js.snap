// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstWallet should render correctly in "debug" mode 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <FirstWallet
    currency="GBP"
    currencyRate=""
    firstAmount={50}
    secondCurrency="USD"
    sign="£"
  >
    <div
      className="wallet-container background"
    >
      <div
        className="d-flex flex-column justify-content-between h-100"
      >
        <div
          className="d-flex justify-content-between align-items-center"
        >
          <Input
            className="wallet-currency background"
            data-testid="wallet-select"
            defaultValue="GBP"
            name="select"
            onChange={[Function]}
            type="select"
          >
            <select
              className="wallet-currency background form-control"
              data-testid="wallet-select"
              defaultValue="GBP"
              name="select"
              onChange={[Function]}
            >
              <option>
                GBP
              </option>
              <option>
                USD
              </option>
              <option>
                EUR
              </option>
            </select>
          </Input>
          <input
            className="wallet-input"
            data-testid="wallet-input-1"
            disabled={false}
            min="0"
            onChange={[Function]}
            type="number"
            value={50}
          />
        </div>
        <div>
          <div
            className="wallet-pocket"
          >
            You Have 
            £
            50.00
          </div>
        </div>
      </div>
    </div>
    <div
      className="arrow-down"
    />
  </FirstWallet>
</Provider>
`;
